client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -20,77 +20,83 @@ import {
 	getTwoFactorPushToken,
 	getTwoFactorUserId,
 } from 'state/login/selectors';
+import { http } from 'state/http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { local } from 'state/data-layer/utils';
 
 /***
  * Module constants
  */
 const POLL_APP_PUSH_INTERVAL_SECONDS = 5;
 
-/***
- * Checks the status of the push notification auth
- *
- * @param {Object}   store  Global redux store
- * @returns {Promise}		Promise of result from the API
- */
-const doAppPushRequest = ( store ) => {
+const requestTwoFactorPushNotificationStatus = ( store, action ) => {
 	const authType = 'push';
 
-	return request.post( 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint' )
-		.withCredentials()
-		.set( 'Content-Type', 'application/x-www-form-urlencoded' )
-		.accept( 'application/json' )
-		.send( {
-			user_id: getTwoFactorUserId( store.getState() ),
-			auth_type: authType,
-			two_step_nonce: getTwoFactorAuthNonce( store.getState(), authType ),
-			remember_me: getRememberMe( store.getState() ),
-			two_step_push_token: getTwoFactorPushToken( store.getState() ),
-			client_id: config( 'wpcom_signup_id' ),
-			client_secret: config( 'wpcom_signup_key' ),
-		} ).then( () => {
-			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_COMPLETED } );
-		} ).catch( error => {
-			store.dispatch( {
-				type: TWO_FACTOR_AUTHENTICATION_UPDATE_NONCE,
-				nonceType: 'push',
-				twoStepNonce: error.response.body.data.two_step_nonce
-			} );
-			return Promise.reject( error );
-		} );
+	store.dispatch(
+		http( {
+			url: 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint',
```

Comment:
That's how I read it...
The query string in this one is part of the endpoint's name, it makes more sense to have it in the URL.

Anyway, added support for query string params in https://github.com/Automattic/wp-calypso/pull/16036/commits/d2dff9535b67f2125103ea52866ff350bd4d7dcf